src-ui/src/app/components/common/profile-edit-dialog/profile-edit-dialog.component.spec.ts

```diff
@@ -265,28 +265,26 @@ describe('ProfileEditDialogComponent', () => {
     expect(getProvidersSpy).toHaveBeenCalled()
   })
 
-  it('should remove disconnected social account from component', async () => {
+  it('should remove disconnected social account from component, show error if needed', async () => {
     const disconnectSpy = jest.spyOn(profileService, 'disconnectSocialAccount')
-    disconnectSpy.mockReturnValue(of(socialAccount.id))
-
-    let resolve
-    const p = new Promise((r) => (resolve = r))
-
     const getSpy = jest.spyOn(profileService, 'get')
-    getSpy.mockImplementation(() => {
-      resolve()
-      return of(profile)
-    })
-
+    getSpy.mockImplementation(() => of(profile))
     component.ngOnInit()
 
-    await p
+    const errorSpy = jest.spyOn(toastService, 'showError')
 
-    expect(getSpy).toHaveBeenCalled()
     expect(component.socialAccounts).toContainEqual(socialAccount)
```

Comment:
Is it guaranteed that the profile observable in `ngOnInit` has populated the `socialAccounts` member already at this point? I thought some synchronization would be needed.

Reply:
For the purposes of this test, it’s fine I think